DigitalLearningSolutions.Data.Tests/Services/EmailServiceTests.cs

```diff
@@ -0,0 +1,296 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.Services
+{
+    using System.Threading;
+    using DigitalLearningSolutions.Data.Factories;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Services;
+    using FakeItEasy;
+    using MailKit.Net.Smtp;
+    using MailKit.Security;
+    using MimeKit;
+    using NUnit.Framework;
+
+    public class EmailServiceTests
+    {
+        private EmailService emailService;
+        private IConfigService configService;
+        private ISmtpClient smtpClient;
+        private Email testEmail;
+        private Email multipleAddresseesTestEmail;
+
+        [SetUp]
+        public void Setup()
+        {
+            BodyBuilder emailTestBody = new BodyBuilder
+            {
+                TextBody = "Test body",
+                HtmlBody = @"<body style= 'font - family: Calibri; font - size: small;'>
+                                    <p>Test Body</p>
+                                </body>"
+            };
+
+            testEmail = new Email(
+                "recipient@example.com",
+                "Test Subject Line",
+                emailTestBody,
+                "cc@example.com",
+                "bcc@example.com");
+
+            multipleAddresseesTestEmail = new Email(
+                new string[2]{"recipient1@example.com", "recipient2@example.com"},
+                "Test Subject Line",
+                emailTestBody,
+                new string[2]{"cc1@example.com", "cc2@example.com"},
+                new string[2]{"bcc1@example.com","bcc2@example.com"});
+
+            configService = A.Fake<IConfigService>();
+            var smtpClientFactory = A.Fake<ISmtpClientFactory>();
+            smtpClient = A.Fake<ISmtpClient>();
+            A.CallTo(() => smtpClientFactory.GetSmtpClient()).Returns(smtpClient);
+
+            A.CallTo(() => configService.GetConfigValue(ConfigService.MailPort)).Returns("25");
+            A.CallTo(() => configService.GetConfigValue(ConfigService.MailUsername)).Returns("username");
+            A.CallTo(() => configService.GetConfigValue(ConfigService.MailPassword)).Returns("password");
+            A.CallTo(() => configService.GetConfigValue(ConfigService.MailServer)).Returns("smtp.example.com");
+            A.CallTo(() => configService.GetConfigValue(ConfigService.MailFromAddress)).Returns("test@example.com");
+
+            emailService = new EmailService(configService, smtpClientFactory);
+            // Going to want to copy the others over, but also perform it with string lists in the to/cc/bcc etc.
+        }
+
+        [Test]
+        public void Trying_to_send_mail_with_null_config_values_should_throw_an_exception()
```

Comment:
Isn't this testing just the MailPassword being missing? Could we potentially use a parameterised test here (I know that in xUnit we could use [Theory], but not sure about nUnit)?

Reply:
Ah, looks like nUnit has [TestCase] syntax